contributed_definitions/nyaml/SPM/NXlockin.yaml

```
@@ -0,0 +1,318 @@
+  
+  # unit could be NX_VOLTAGE or NX_CURRENT
+  reference_amplitude(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      Amplitude of the reference signal for the lock-in amplifier.
+  reference_frequency(NX_NUMBER):
+    unit: NX_FREQUENCY
+    doc: |
+      Frequency of the reference signal for the lock-in amplifier.
+  reference_phase(NX_NUMBER):
+    unit: NX_ANGLE
+    doc: |
+      Frequency of the reference signal for the lock-in amplifier.
+  demodulated_signal(NX_CHAR):
+    doc: |
+      The modulated output signal will be demodulated, in order to determine the amplitude
+      and phase at the frequency set in the Frequency field or harmonics, such as current,
+      bias, et.al.
+  demodulated_frequency(NX_NUMBER):
+    unit: NX_FREQUENCY
+    doc: |
+      The frequency of the demodulated signal.
+  frequency_demodulation_bandwidth(NX_NUMBER):
+    unit: NX_FREQUENCY
+    doc: |
+      The bandwidth of the modulated signal that can be applied in frequency
+      demodulation mechanism.
+  
+  # unit could be NX_VOLTAGE or NX_CURRENT
+  demodulated_amplitude(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      The amplitude of the demodulated signal.
+  amplitude_demodulation_bandwidth(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      The bandwidth of the modulated signal that can be applied in amplitude
+      demodulation mechanism.
+  demodulated_phase(NX_NUMBER):
+    unit: NX_ANGLE
+    doc: |
+      The phase of the demodulated signal.
+  phase_demodulation_bandwidth(NX_NUMBER):
+    unit: NX_FREQUENCY
+    doc: |
+      The bandwidth of the modulated signal that can be applied in phase demodulation
+      mechanism.
+  demodulator_channels(NX_CHAR):
+    doc: |
+      List of the demodulator channels.
+  low_pass_N(NX_NUMBER):
+    unit: NX_FREQUENCY
+    doc: |
+      Frequency of the low-pass filter applied on the demodulated
+      signals (X,Y), cut-off frq (low pass filter) (for each DemodulatorChannels).
+  hi_pass_N(NX_NUMBER):
+    unit: NX_FREQUENCY
+    doc: |
+      Frequency of the high-pass filter applied on the demodulation
+      signal cut-off frq (hi pass filter) (for each DemodulatorChannels).
+  lp_filter_order_N(NX_NUMBER):
+    doc: |
+      Order of the low-pass filter applied on the demodulated signals (X, Y).
+      Reducing the bandwidth or increasing the order reduces noise on the demodulated signals,
+      but will require longer settling and measurement times.
+  hp_filter_order_N(NX_NUMBER):
+    doc: |
+      Order of the high-pass filter applied on the demodulation
+      signal. This is used mainly to suppress a DC component of the input
+      signal noise.
+  ref_phase_N(NX_NUMBER):
+    unit: NX_ANGLE
+    doc: |
+      Reference phase of reference signal with respect to the demodulated signal
+      (foreach Channels).
+  integration_time(NX_NUMBER):
+    unit: NX_TIME
+    doc: |
+      Time during which the data are acquired and averaged. (for the input filter)
```

Integration time is not a parameter of the filters

```suggestion
      Integration time for the product of the input and the reference signals
```